recipe/meta.yaml

```diff
@@ -27,13 +27,68 @@ source:
     folder: testing
 
 build:
-  number: 4
+  number: 5
   # for cuda support, building with one version is enough to be compatible with
   # all later versions, since arrow is only using libcuda, and not libcudart.
   skip: true  # [cuda_compiler_version not in ("None", cuda_compiler_version_min)]
   # arrow promises API- & ABI-compatibility along SemVer, see #1096
   run_exports:
-    - {{ pin_subpackage("libarrow", max_pin="x") }}
+    - {{ pin_subpackage("libarrow-all", max_pin="x") }}
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ compiler("cuda") }}                 # [cuda_compiler_version != "None"]
+    # needs to run protoc & grpc_cpp_plugin
+    - libgrpc                                # [build_platform != target_platform]
+    - libprotobuf                            # [build_platform != target_platform]
+    # needed for gandiva
+    - clangdev {{ llvm_version }}            # [build_platform != target_platform]
+    - llvmdev {{ llvm_version }}             # [build_platform != target_platform]
+    - gnuconfig                              # [build_platform != target_platform]
+    - cmake
+    - ninja
+    # necessary for vendored jemalloc
+    - autoconf     # [linux]
+    - make         # [linux]
+  host:
+    # for required dependencies, see
+    # https://github.com/apache/arrow/blob/apache-arrow-11.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake#L46-L75
+    - clangdev {{ llvm_version }}
+    - llvmdev {{ llvm_version }}
+    - aws-crt-cpp
+    - aws-sdk-cpp
+    - boost-cpp >=1.70
+    - brotli
+    - bzip2
+    # not yet: https://github.com/conda-forge/cpp-opentelemetry-sdk-feedstock/issues/38
+    # - cpp-opentelemetry-sdk
+    # - proto-opentelemetry-proto =={{ cpp_opentelemetry_sdk }}
+    - gflags
+    - glog
+    - google-cloud-cpp
+    # arrow uses a customized jemalloc, see #944
+    # - jemalloc
+    - libabseil
+    - libutf8proc
+    - lz4-c
+    - nlohmann_json
+    - orc
+    - rapidjson
+    - re2
+    - snappy
+    - thrift-cpp
+    - ucx          # [linux]
+    - xsimd
+    - zlib
+    - zstd
+    - __cuda >=11.2  # [cuda_compiler_version != "None"]
```

Comment:
@h-vetinari this was raising an error locally if I used here `__cuda >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]`. The error when running `python build-locally.py linux_64_cuda_compiler_versionNone`:
```
RuntimeError: Received dictionary as spec.  Note that pip requirements are not supported in conda-build meta.yaml.  Error message: Invalid version '>=': invalid operator
```
Do you know if there is any issue where the global host dependency syntax would behave differently with the templating syntax?